echopype/echodata/zarr_combine.py

```diff
@@ -0,0 +1,742 @@
+        this group specific information to the class variable
+        ``group_attrs``.
+
+        Parameters
+        ----------
+        ds_list: List[xr.Dataset]
+            The Datasets that will be combined
+        ed_name: str
+            The name of the EchoData group corresponding to the
+            Datasets in ``ds_list``
+
+        Notes
+        -----
+        This method creates the following class variables:
+        dims_df: pd.DataFrame
+            Dataframe with column as dim names, rows as the
+            different Datasets, and values as the length of the dimension
+        dims_sum: dict
+            Keys as the dimension name and values as the corresponding
+            sum of the lengths across all Datasets
+        dims_csum: dict
+            Keys as the dimension name and values as a dictionary of
+            the corresponding cumulative sum of the lengths across
+            all Datasets
+        dims_max: dict
+            Keys as the dimension name and values as the corresponding
+            maximum length across all Datasets
+
+        If attribute values are numpy arrays, then they will not be included
+        in the ``self.group_attrs``. Instead, these values will only appear
+        in the attributes of the combined ``EchoData`` object.
+        """
+
+        self._check_ascending_ds_times(ds_list, ed_name)
+        self._check_channels(ds_list, ed_name)
+
+        # Dataframe with column as dim names and rows as the different Datasets
+        self.dims_df = pd.DataFrame([ds.dims for ds in ds_list])
+
+        # calculate useful information about the dimensions
+        self.dims_sum = self.dims_df.sum(axis=0).to_dict()
+        self.dims_csum = self.dims_df.cumsum(axis=0).to_dict()
+        self.dims_max = self.dims_df.max(axis=0).to_dict()
+
+        # format ed_name appropriately
+        ed_name = ed_name.replace("-", "_").replace("/", "_").lower()
+
+        if len(ds_list) == 1:
+            # get numpy keys if we only have one Dataset
+            numpy_keys = self._compare_attrs(ds_list[0].attrs, ds_list[0].attrs)
+        else:
+            # compare attributes and get numpy keys, if they exist
+            for ind in range(len(ds_list) - 1):
+                numpy_keys = self._compare_attrs(ds_list[ind].attrs, ds_list[ind + 1].attrs)
+
+        # collect Dataset attributes
+        for count, ds in enumerate(ds_list):
+
+            # get reduced attributes that do not include numpy keys
+            red_attrs = {key: val for key, val in ds.attrs.items() if key not in numpy_keys}
+
+            if count == 0:
+                self.group_attrs[ed_name + "_attr_key"].extend(red_attrs.keys())
+            self.group_attrs[ed_name + "_attrs"].append(list(red_attrs.values()))
+
+    def _get_temp_arr(self, dims: List[str], dtype: type) -> Tuple[type(dask.array), list]:
+        """
+        Constructs a temporary (or dummy) array representing a
+        variable in its final combined form.
```

Comment:
might be useful to add something about the array has the shape of the final combined data

Reply:
Sounds good, I will add another sentence stating this.